src/index.js

```diff
@@ -25,31 +25,43 @@ function buildSchema(swaggerPath, options) {
     return Promise.all([
         SwaggerParser.dereference(swaggerPath),
         SwaggerParser.parse(swaggerPath)
-    ]).then(function ([dereferenced, referenced]) {
-        return buildValidations(referenced, dereferenced, options);
+    ]).then(function ([dereferencedJsonSchema, jsonSchema]) {
+        return buildValidations(jsonSchema, dereferencedJsonSchema, options);
     });
 }
 
-function buildSchemaSync(swaggerPath, options) {
-    const parsed = yaml.load(fs.readFileSync(swaggerPath), 'utf8');
-    const dereferenced = deref(parsed);
+function buildSchemaSync(pathOrSchema, options) {
+    const jsonSchema = getJsonSchema(pathOrSchema);
+    const dereferencedSchema = deref(jsonSchema);
 
-    return buildValidations(parsed, dereferenced, options);
+    return buildValidations(jsonSchema, dereferencedSchema, options);
+}
+
+function getJsonSchema(pathOrSchema) {
+    if (pathOrSchema instanceof Object) {
```

Comment:
Isn't it easier to check if `pathOrSchema` is string? Judging by Lodash implementation, check is not that simple and there might be edge cases (https://github.com/lodash/lodash/blob/master/isObject.js)

Reply:
fixed